app/analyzers/sudden_appearance.py

```diff
@@ -0,0 +1,144 @@
+from helpers.singletons import settings, es
+from helpers.analyzer import Analyzer
+import datetime as dt
+import dateutil.parser
+
+
+class SuddenAppearanceAnalyzer(Analyzer):
+
+    def __init__(self, model_name, config_section):
+        super(SuddenAppearanceAnalyzer, self).__init__("sudden_appearance", model_name, config_section)
+
+    def _extract_additional_model_settings(self):
+        """
+        Override method from Analyzer
+        """
+        self.model_settings["timestamp_field"] = settings.config.get("general",
+                                                                     "timestamp_field",
+                                                                     fallback="@timestamp")
+
+        self.model_settings["target"] = self.config_section["target"].replace(' ', '').split(",")
+        self.model_settings["aggregator"] = self.config_section["aggregator"].replace(' ', '').split(",")
+
+        self.model_settings["slide_window_days"] = self.config_section.getint("slide_window_days")
+        self.model_settings["slide_window_hours"] = self.config_section.getint("slide_window_hours")
+        self.model_settings["slide_window_mins"] = self.config_section.getint("slide_window_mins", fallback=0)
+        self.model_settings["slide_jump_days"] = self.config_section.getint("slide_jump_days")
+        self.model_settings["slide_jump_hours"] = self.config_section.getint("slide_jump_hours")
+        self.model_settings["slide_jump_mins"] = self.config_section.getint("slide_jump_mins", fallback=0)
+
+        self.model_settings["trigger_slide_window_proportion"] = self.config_section.getfloat(
+            "trigger_slide_window_proportion")
+
+        self.delta_history_win = dt.timedelta(days=self.model_settings["history_window_days"],
```

Comment:
This and the next parts seem like a lot of duplication of code, which could be grouped into a method.